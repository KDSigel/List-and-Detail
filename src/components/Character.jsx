export default function Character({character}) {

    return (
        <>
            <img src={character.photoUrl} alt={character.name} width="500" />
            <div>name: {character.name}</div>
            <div>position: {character.position}</div>
            <div>allies: {character.allies}</div>
        </>
    )
}
