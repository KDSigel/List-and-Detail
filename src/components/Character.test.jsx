import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Character from './Character'

it('tests a snapshot of Character component', () => {

const character = {
    name: 'karl',
    position: 'driver',
    allies: 'cohort',
    photoUrl: 'https://en.wikipedia.org/wiki/Vasily_Arkhipov#/media/File:Vasili_Arkhipov_young.jpg'
}

const {container} = render(
    <MemoryRouter>
        <Character character={character}/>
    </MemoryRouter>

)

expect(container).toMatchSnapshot()
}
)