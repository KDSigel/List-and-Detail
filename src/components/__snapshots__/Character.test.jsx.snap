// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests a snapshot of Character component 1`] = `
<div>
  <img
    alt="karl"
    src="https://en.wikipedia.org/wiki/Vasily_Arkhipov#/media/File:Vasili_Arkhipov_young.jpg"
    width="300"
  />
  <div>
    name: 
    karl
  </div>
  <div>
    position: 
    driver
  </div>
  <div>
    allies: 
    cohort
  </div>
</div>
`;
